### PR TITLE
Camera: Add support for video capture and range zoom

### PR DESCRIPTION
**Pull request by DonLakeFlyer**
This helps me a ton with testing QGC camera code:
* Added support for range zoom
* Added support for video capture

### CODE REVIEW

File: src/gazebo_camera_manager_plugin.cpp [hunk 1]
```diff
@@ -540,6 +545,55 @@ void CameraManagerPlugin::_handle_stop_take_photo(const mavlink_message_t *pMsg,
...
+{
+    gzdbg << "Handle Stop Video Capture" << endl;
+
+    mavlink_command_long_t cmd;
+    mavlink_msg_command_long_decode(pMsg, &cmd);
+    std::lock_guard<std::mutex> guard(_captureMutex);
+
+    if (cmd.param1 != 0) {
+        gzerr << "VIDEO_STOP_CAPTURE: param1 must be 0\n";
+        _send_cmd_ack(pMsg->sysid, pMsg->compid, MAV_CMD_VIDEO_START_CAPTURE, MAV_RESULT_UNSUPPORTED, srcaddr);
```
Comment (julianoes) [inline src/gazebo_camera_manager_plugin.cpp]: Same here, see below.

File: src/gazebo_camera_manager_plugin.cpp [hunk 2]
```diff
@@ -540,6 +545,55 @@ void CameraManagerPlugin::_handle_stop_take_photo(const mavlink_message_t *pMsg,
...
+
+    if (_captureMode != CAPTURE_DISABLED) {
+        // We are already capturing
+        _send_cmd_ack(pMsg->sysid, pMsg->compid, MAV_CMD_VIDEO_START_CAPTURE, MAV_RESULT_TEMPORARILY_REJECTED, srcaddr);
+        return;
+    }
+
+    if (cmd.param1 != 0 || cmd.param2 != 0) {
+        gzerr << "VIDEO_START_CAPTURE: param1 and param2 must be 0\n";
+        _send_cmd_ack(pMsg->sysid, pMsg->compid, MAV_CMD_VIDEO_START_CAPTURE, MAV_RESULT_UNSUPPORTED, srcaddr);
```
Comment (julianoes) [inline src/gazebo_camera_manager_plugin.cpp]: [MAV_RESULT_UNSUPPORTED](https://mavlink.io/en/messages/common.html#MAV_RESULT_UNSUPPORTED). is to be used when the whole command is not supported, I think.

In this case I would say [MAV_RESULT_DENIED](https://mavlink.io/en/messages/common.html#MAV_RESULT_DENIED).

**Review by julianoes — COMMENTED** [review]
Nice! See comments inline.

File: src/gazebo_camera_manager_plugin.cpp (hunk 2 above)
Comment (DonLakeFlyer) [reply src/gazebo_camera_manager_plugin.cpp]: Ah, was duplicating how _handle_take_photo handles basically the same thing. Should I change that code as well to MAV_RESULT_DENIED?

Comment (DonLakeFlyer) [reply src/gazebo_camera_manager_plugin.cpp]: You can see it here: https://github.com/PX4/PX4-SITL_gazebo-classic/blob/main/src/gazebo_camera_manager_plugin.cpp#L526

Comment (DonLakeFlyer) [reply src/gazebo_camera_manager_plugin.cpp]: I switched the original code as well

File: src/gazebo_camera_manager_plugin.cpp (hunk 1 above)
Comment (DonLakeFlyer) [reply src/gazebo_camera_manager_plugin.cpp]: Done

**Review by julianoes — APPROVED** [review]

**Comment by DonLakeFlyer** [comment]
No idea why various CI tests are failing. Seems unrelated?

**Comment by julianoes** [comment]
Yes, doesn't look related.

**Comment by DonLakeFlyer** [comment]
Thanks Julian

**Pull request merged**